### PR TITLE
pretome: add imdb search and code clean up

### PR DESCRIPTION
**Pull request by ngosang**
Tested. Part of #4859

### CODE REVIEW

**Comment by ngosang** [comment]
@garfield69 I'm having a problem with this. If you take a look => https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/Pretome.cs#L101, they are repeating all categories to fix  this problem.

If I select the category TV, the search is ok, but the response won't include the TV/HD, TV/SD... just TV. Adding all categories again to the mapping (with the parent category) works, but must be a better solution....

**Comment by ngosang** [comment]
@RoloSoze ping

**Comment by garfield69** [comment]
yes, I think its because the code is using **string** custom categories, so Torznab does not do the subcat to parent cat association that would happen if the code had been using custom **numbers**, where, for instance, a category number  of 15 for Audio/Video for example,  gets turned by Torznab into custom category number 100015, and automatically gets association with Torznab subcategory 3020 and parent category 3000.
But going the custom number route for this particular site would have necessitated a **case** block in the results processing section, in order to match the results selectors string back up with a category number, so I think your latest improvement is a good compromise.

File: src/Jackett.Common/Indexers/Pretome.cs [hunk 1]
```diff
@@ -188,93 +133,78 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-                queryCollection.Add("search", query.GetQueryString());
+                qc.Add("search", query.GetQueryString());
+                qc.Add("st", "1");
             }
 
-            // Do not include too many tags as it'll mess with their servers.
-            if (tags.Split(',').Length < 7)
+            // parse categories and tags
+            var catGroups = new List<string>();
+            var tagGroups = new List<string>();
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/Pretome.cs]: If we're going to call distinct at the end anyway, might as well make this `HashSet<string>` and remove lines 185&186. HashSet and List behave the same in the remaining code.

Comment (ngosang) [reply src/Jackett.Common/Indexers/Pretome.cs]: done

**Comment by ngosang** [comment]
@cadatoiva I'm going to merge, we have to open another PR after #7915 to replace `List<KeyValuePair<string, string>>` in other trackers too.

**Pull request merged**